tests/unit/panther_analysis_tool/test_main.py

```diff
@@ -62,6 +62,13 @@ def test_with_filters(self):
         assert_equal(return_code, 0)
         assert_equal(len(invalid_specs), 0)
 
+    def test_invalid_rule_definition(self):
+        args = pat.setup_parser().parse_args('test --path tests/fixtures --filter Severity=Critical RuleID=AWS.CloudTrail.MFAEnabled'.split())
+        args.filter = pat.parse_filter(args.filter)
+        return_code, invalid_specs = pat.test_analysis(args)
+        assert_equal(return_code, 1)
+        assert_equal(len(invalid_specs), 1)
```

Comment:
I think we want  `len(invalid_specs)` to be 0 here. `invalid_specs` should only be populated if there are any rules/policies that have invalid specification files.  In this case, I think the specification file should be valid but the tests should fail, which should have a `return_code` of 1 and no `invalid_specs`.

Reply:
I tried to create a filter (Severity=Critical RuleID=AWS.CloudTrail.MFAEnabled) for that test that would only pick up the new rule files, but it would continue to show the invalid spec for the file in the same directory (example_malformed_policy.yml). I can remove that check altogether since that particular check is done in a different test case, or is there a different way to filter to just pick up those specific, new files?

Reply:
Ideally we would just filter to the new files. You've already created a brand new rule body & specification file, so you can just update the `RuleID` for your new rule to be something like `AWS.CloudTrail.MFAEnabled.ReturnsNone`, then use that as your filter. That should filter out to just the test cases you care about.

Reply:
I updated the filter, but it was still returning an invalid spec from the directory:

```
% panther_analysis_tool test --path tests/fixtures --filter RuleID=AWS.CloudTrail.MFAEnabled.ReturnsNone
[INFO]: Testing analysis packs in tests/fixtures

AWS.CloudTrail.MFAEnabled.ReturnsNone
        [FAIL] User MFA enabled passes compliance but fails dedup check.
                [FAIL] [dedup] None
                [PASS] [title] None does not have MFA enabled

--------------------------
Panther CLI Test Summary
        Path: tests/fixtures
        Passed: -1
        Failed: 1
        Invalid: 1
```

It looks like the invalid spec check (classify_analysis, line 246) is done prior to the filtering (filter_analysis, line 254), so any invalid spec in the directory will still be flagged in the output? Since this is the failure case, I didn't want to add it to the "valid_analysis" dir.

Reply:
Got it. This should be fine then, thanks for looking into it.